roles/couchbase-infra/tasks/main.yml

```diff
@@ -0,0 +1,206 @@
+---
+
+# Install Couchbase Operator (Assume rook is already in place? or should we check?)
+- name: Create ClusterRole for the Benchmark User
+  k8s:
+    definition: "{{ lookup('template', 'cluster-role-benchmark.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRoleBinding for the Benchmark User
+  k8s:
+    definition: "{{ lookup('template', 'cluster-role-binding-benchmark.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Create CRD for Couchbase Operator
+  k8s:
+    definition: "{{ lookup('file', 'crd.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Reticulating Splines
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ couchbase_namespace }}"
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRole for Service Account
+  k8s:
+    definition: "{{ lookup('file', 'cluster-role-sa.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Create Service Account
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: ServiceAccount
+      metadata:
+        name: couchbase-operator
+        namespace: "{{ couchbase_namespace }}"
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRoleBinding for the Service Account
```

Comment:
Since the operator creates these, we need to add resiliency so that the operator can also delete these when we set couchbase.servers.size to 0 which would also mean that the operator will have to track everything it has created. Another way would be if we ask user to perform all this.

Reply:
I think "cleanup" is a general issue we need to address across roles. Things _do_ cleanup if you simply delete the CR, but getting a job to cleanup when it completes execution is a different story.

Reply:
So I guess the question is, should we resolve cleanup for this workload before merging it to master, or should cleanup be a project-level issue to be tracked?

Reply:
that is true, but if we take up uperf role, the creation of rbac and such is done outside the scope of operator so it becomes the user's responsibility to clean up the rbac and such. but in this case we're creating it inside the operator so lets say if we instead of deleting cr apply new change so that after ycsb uperf is run, it will create an issue I feel. I've not been able to test it out so far so I don't know the in and outs.

Reply:
@aakarshg you bring up a good point.

So, either we assume the user adds the Couchbase RBAC, which will be around if they run UPerf/FIO/etc... Or we create the RBAC for them via the operator, in which case, it will be around as long as the CR exists..

I would prefer we have a all-in-one solution, where we simply pull in the Couchbase Operator, and launch infrastructure... Making automation much cleaner. However, automating the RBAC creation prior to running Couchbase would be trivial.

Reply:
As discussed in meetings/slack, are we OK with moving this out to a project-level issue to be tracked since a decision here isn't clear and will likely affect other infra/workload additions in the future?